### ISSUE TITLE
notmuch as virtual-mailbox is counting wrong

### ISSUE THREAD

**Issue by therealmaxmoon** [neomutt/neomutt#2123]
## Expected Behaviour

Show all emails, which are in the inbox (unread/total).

## Actual Behaviour

It shows all emails, even those, which are in the trash.

## Steps to Reproduce

I used the tutorial on [neomutt.org](https://neomutt.org/feature/notmuch) to add a virtual mailbox with notmuch like this:
`virtual-mailboxes "My INBOX" "notmuch://?query=tag:inbox"`
But no matter how many messages I delete (move to trash). The total amount of mails stays the same. With a normal mailbox (=INBOX) the amount is right.

Or is moving the email to trash not enough? Is the proper way to tag it as trash? What would be the right tag?

## How often does this happen?

- Always

## When did it start to happen?

- Since I use virtual mailboxes with notmuch

## NeoMutt Version

```
NeoMutt 20191111
```

**Comment by Austin-Ray** [comment]
Are you removing the `inbox` tag when moving to email to the trash folder? The virtual mailbox will return all mail matching the query.

**Comment by therealmaxmoon** [comment]
No, I am not removing the inbox tag and I am a little bit afraid to ask how to do it properly. 
The only thing that happens is:
`bind index,pager dd delete-message`
and this will be executed in a folder-hook:
`set trash = "+my_mailbox/Trash"`
And maybe it could solve this problem too, if there is a better/proper way to do it #2127

**Comment by therealmaxmoon** [comment]
And shouldn't be there an added tag like "trash" or is everything automatically trash, which has no inbox tag? Sorry, I am really new to neomutt and notmuch and I am trying to work hard every day to understand it. But sometimes I just can't see obvious things, I guess...

**Comment by Austin-Ray** [comment]
> I am a little bit afraid to ask how to do it properly.
> I  am really new to neomutt and notmuch

Don't worry about! Everyone was a newbie at once. You'll get there :)

So notmuch doesn't have the concept of deleting messages instead it uses exclusions[0]. They're turned off by default according to that documentation so you can turn them on for the `notmuch` CLI. 

However, that doesn't seem to affect Neomutt's queries, but fortunately there's the `nm_exclude_tags` configuration option[1] where you set them. Then all queries will ignore messages with those tags __unless__ you specifically query the excluded tag.

It looks like deleting a message in a notmuch mailbox will automatically add the `deleted` tag (I don't delete mail so today I learned!) Then if `deleted` is an excluded tag, the count should be accurate. 

[0] https://notmuchmail.org/excluding/
[1] https://neomutt.org/feature/notmuch#4-%C2%A0variables

**Comment by therealmaxmoon** [comment]
I put the nm_exlude_tags like this in the neomutt config file:

`set nm_exclude_tags = deleted`

I've tried to tag messages with the "deleted" flag, like you described it here: https://github.com/neomutt/neomutt/issues/1736

```
bind index,pager ff noop
macro index,pager ff "<modify-tags-then-hide>+deleted<enter>" "delete message"
```
But if I press ff, neomutt just says: Key is not bound.  Press '?' for help.
But in the help page, it shows the macro for ff.
And if I search for taged messages with notmuch, no results are there:
`notmuch search tag:deleted`

**Comment by Austin-Ray** [comment]
You actually don't need a custom macro to add the deleted tag. Neomutt will add it automatically if you delete an email from a notmuch mailbox (didn't know that when i initially commented.)

**Comment by therealmaxmoon** [comment]
It doesn't work for me.

I have mixed mailboxes. One is a virtual mailbox for all unread emails and the other ones are real directories.

Keybinding to delete a message is:
`bind index,pager dd delete-message`

After hitting dd in the virtual mailbox and closing Neomutt, the following message pops up:
`Purge 1 deleted message? ([yes]/no):`
I pressed 'y' and tried to look for the deleted tag with notmuch:
`notmuch search tag:deleted`
But nothing was found. But the deleted message can be found in the trash directory of the email account.

First I thought it's because I've excluded the deleted tag in notmuch before, but the [page](https://notmuchmail.org/excluding/) says, if you search for the tag explicitly, it should appear.

**Comment by Austin-Ray** [comment]
Ah, it appears I've overlooked two lines in my increasingly disorganized configurations files. I _do_ have two macros to support deletion after all. I don't regularly use non-notmuch mailboxes so I can't guarantee they'll work 100% out of the box for other mailbox kinds.

```neomuttrc
macro index d "<delete-message><modify-tags-then-hide>+deleted<return>k"
macro index u "<undelete-message><modify-tags-then-hide>-deleted<return>k"
```

You might want to use a `folder-hook` to apply this only to notmuch mailboxes using the regex `notmuch://*`

My apologies for overlooking those macros.

**Comment by therealmaxmoon** [comment]
Ok, I have a really weird behavior now. 

If I press d to delete a message in a virtual mailbox with the macro above, it gets a **D** and after syncing neomutt the message will not be seen in the current view. Now I check if the message was tagged like it should be and I set up a virtual mailbox like this and navigate to it:
`virtual-mailboxes "Deleted" "notmuch://?query=tag:deleted"`
The weird part is, that in the deleted mailbox is not the message I deleted, it is the message after the one I've deleted. And if I navigate back to the virtual mailbox, the message I wanted to delete appears again, but the message, which was directly behind this message is gone (deleted).

The virtual mailbox I deleted a message from is:
`virtual-mailboxes "Unread" "notmuch://?query=tag:unread"`

I can't explain what's happening here?

**Comment by therealmaxmoon** [comment]
I found the solution. After `<delete-message>` the cursor jumps to the next message in list and with `<modify-tags-then-hide>+deleted` the wrong message will be tagged. So the solution might be going one message up after deletion (like @Austin-Ray did it with the k in the end):

```
macro index dd "<delete-message><previous-entry><modify-tags-then-hide>+deleted<return><previous-entry>"
macro index u "<undelete-message><previous-entry><modify-tags-then-hide>-deleted<return><previous-entry>"
```
But this solution has a little issue. It will not work for the last message in the list, because the cursor jumps one message up and tags the message before the one, which will be deleted. 

So I guess mixing up real and virtual mailboxes is just not a good idea. 

**My solution:** I will change my mailboxes to pure virtual mailboxes and the hard deletion of the emails will be the task of an external program.

**Comment by dikeert** [comment]
A little note to whoever comes here later.

You don't need `<previous-entery>` or `k` like @Austin-Ray  has, instead you need to have this:
```
: unset resolve\n
```
as first thing in your macro and that:
```
: set resolve\n
```
as the last one to enable it back.

**Comment by marvinwankersteen** [comment]
I stumbled across exactly the same behavior as @therealmaxmoon explained in https://github.com/neomutt/neomutt/issues/2123#issuecomment-593874008.

Thank you @tnymlr for this solution. I have been looking for such a setting, but had not found anything suitable, until now ;-)

I have recently switched to neomutt and would like to maintain the real mailboxes, at least in part, in parallel to the virtual mailboxes. At the very least, I would like to actually move the deleted mails to the real trash.
